# Домашнее задание №8

**Задание 1**

Определите, какому фильму было выставлено больше всего оценок 5.0.

In [2]:
#импорт необходимых библиотек
import pandas as pd

In [3]:
#загрузка данных из CSV файлов
movies_df = pd.read_csv('movies.csv')
ratings_df = pd.read_csv('ratings.csv')

In [4]:
#знакомство с данными
print("Первые 5 строк таблицы movies:")
print(movies_df.head())

print("\nПервые 5 строк таблицы ratings:")
print(ratings_df.head())

Первые 5 строк таблицы movies:
   movieId                               title  \
0        1                    Toy Story (1995)   
1        2                      Jumanji (1995)   
2        3             Grumpier Old Men (1995)   
3        4            Waiting to Exhale (1995)   
4        5  Father of the Bride Part II (1995)   

                                        genres  
0  Adventure|Animation|Children|Comedy|Fantasy  
1                   Adventure|Children|Fantasy  
2                               Comedy|Romance  
3                         Comedy|Drama|Romance  
4                                       Comedy  

Первые 5 строк таблицы ratings:
   userId  movieId  rating  timestamp
0       1        1     4.0  964982703
1       1        3     4.0  964981247
2       1        6     4.0  964982224
3       1       47     5.0  964983815
4       1       50     5.0  964982931


**Примечание:** В таблицах есть "объединяющий" параметр movieId

In [7]:
#отфильтруем фильмы и выведем только те, у которых рейтин = 5
five_stars = ratings_df[ratings_df['rating'] == 5.0]
five_stars.head() #проверка работы фильтра

userId  movieId  rating  timestamp
3       1       47     5.0  964983815
4       1       50     5.0  964982931
6       1      101     5.0  964980868
8       1      151     5.0  964984041
9       1      157     5.0  964984100

In [18]:
#посчитаем, сколько 5-балльных оценок у каждого фильтра, сгруппировав их по ID
five_stars_count = five_stars.groupby('movieId').size()
five_stars_count.sort_values(ascending=False) #проверка работы группировки + cортировка по убыванию

movieId
318       153
296       123
356       116
2571      109
260       104
         ... 
6732        1
6731        1
6713        1
2066        1
187717      1
Length: 2954, dtype: int64

**Промежуточный вывод:** Фильм с ID = 318 получил больше всего оценок 5.0 (а именно 153 штуки)

In [21]:
# найдем фильм с максимальным количеством оценок 5.0
best_movie_id = five_stars_count.idxmax()
print(max_five_star_movie_id)

318


In [26]:
#найдем название фильма по movieId
best_movie_title = movies_df[movies_df['movieId'] == best_movie_id]['title'].values[0]

#выведем название фильма
print(f"\nФильму '{best_movie_title}' было выставлено больше всего оценок 5.0")


Фильму 'Shawshank Redemption, The (1994)' было выставлено больше всего оценок 5.0


**Задание 2**

По данным файла power.csv посчитайте суммарное потребление стран Прибалтики (Латвия, Литва и Эстония) категорий 4, 12 и 21 за период с 2005 по 2010 год. **Не учитывайте в расчётах отрицательные значения quantity!**

In [32]:
#загрузка данных из CSV файла
power_df = pd.read_csv('power.csv')

In [33]:
#знакомство с данными
print(power_df.head())

   country  year  quantity  category
0  Austria  1996       5.0         1
1  Austria  1995      17.0         1
2  Belgium  2014       0.0         1
3  Belgium  2013       0.0         1
4  Belgium  2012      35.0         1


In [38]:
#отфильтруем данные по странам Прибалтики и категориям
baltic_countries = ['Latvia', 'Lithuania', 'Estonia']
categories = [4, 12, 21]

filtered_df = power_df[
    (power_df['country'].isin(baltic_countries)) &
    (power_df['category'].isin(categories)) &
    (power_df['year'].between(2005, 2010)) &
    (power_df['quantity'] >= 0)  #учитываем только положительные значения
]
filtered_df.head()

country  year  quantity  category
10617    Estonia  2008       1.0         4
10618    Estonia  2007       0.0         4
10619    Estonia  2006       0.0         4
10620    Estonia  2005       1.0         4
11156  Lithuania  2008       1.0         4

In [39]:
#cуммируем данные по потреблению
total = filtered_df['quantity'].sum()

#выведем результат
print(f"\nСуммарное потребление стран Прибалтики (Латвия, Литва, Эстония) категорий 4, 12 и 21 за период с 2005 по 2010 год: {total}")


Суммарное потребление стран Прибалтики (Латвия, Литва, Эстония) категорий 4, 12 и 21 за период с 2005 по 2010 год: 240580.0
